src/modengine/ext/mod_loader/wwise_file_overrides.cpp

```diff
@@ -0,0 +1,120 @@
+// EBLFileOperator, this implementation does all its fetching from the BDTs.
+// EBLFileOperator does not use the usual virtual path lookup that ME2 already 
+// hooks. Hence we need this hook here.
+//
+// In order to selectively make it read from disk again this hook checks if an 
+// override file exists and sets the openMode back from 9 to Read and replaces
+// the virtual path parameter string with an absolute path.
+//
+// This is a messy one though:
+// Figuring out if there is an override isn't straight forward. The hooked 
+// function gets invoked with a virtual path string ex: `sd:/50846376.wem`. 
+// Wwise uses subdirectories for localized content, meaning that the "simple"
+// WEM example makes Wwise look in `/50846376.wem` but also in 
+// `enus/50846376.wem` (or `ja/50846376.wem` for AC6 which has `ja` as an extra 
+// locale for some audio and BNKs). To make matters even worse, Elden Ring 
+// specifically sorts the WEMs into a subdir (`wem/`) and then another subdir 
+// based on the  first two digits of the WEM. So above example will spawn 
+// lookups in  `/wem/50/50846376.wem` and `enus/wem/50/50846376.wem`. In order 
+// to figure out if there is an override we will need to look in multiple 
+// directories per request. Luckily, aside from boot, this routine is called
+// quite infrequently.
+//
+// Also, worth pointing out that not all paths passed to this hook will have 
+// the `sd:/` prefix. So we cannot get away with the usual prefix / rewrite 
+// trick and will have to allocate a completely new string.
+
+namespace fs = std::filesystem;
+
+using namespace spdlog;
+
+std::optional<fs::path> find_override_file(const fs::path game_path)
+{
+    for (const auto& root : hooked_file_roots) {
+        trace(L"Searching for {} in {}", game_path.wstring(), root);
+
+        auto file_path = root / fs::path(game_path);
+        if (fs::exists(file_path)) {
+            return file_path;
+        }
+    }
+
+    return {};
+}
+
+const wchar_t* prefixes[3] = {
+    L"sd/",
+    L"sd/enus/",
+    L"sd/ja/",
+};
+
+std::optional<fs::path> check_paths(const std::wstring filename) {
+    for (auto prefix: prefixes) {
+        if (auto override = find_override_file(prefix+filename)) {
+            return override;
+        }
+    }
+
+    return {};
+}
+
+std::optional<fs::path> find_override(const std::wstring filename)
+{
+    // Check wem/<first to digits of filename>/<filename> too since ER uses 
+    // this format
+    if (filename.ends_with(L".wem")) {
+        auto wem_path = L"wem/" + filename.substr(0,2) + L"/" + filename;
+        auto wem_path_result = check_paths(wem_path);
+
+        if (wem_path_result.has_value()) {
+            return wem_path_result;
+        }
+    }
+
+    return check_paths(filename);
+}
+
+std::optional<std::wstring> normalize_filename(const std::wstring path) {
+    if (path.length() > 3 && path.substr(0, 4) == L"sd:/") {
```

Comment:
Can this be `path.starts_with(...)` from C++20? I think ME2 is targeting C++20 nowadays.

Reply:
Should be resolved per 8a857f895fe20aedaf5fd9357df2b9013b6d7277